### PR TITLE
Fix some compiling warnings

### PR DESCRIPTION
**Pull request by bukepo**
This PR simply fixes some annoying warnings.

### CODE REVIEW

File: src/wpanctl/tool-cmd-commissioner.c [hunk 1]
```diff
@@ -305,7 +305,7 @@ int tool_cmd_commissioner(int argc, char* argv[])
 					goto bail;
 				}
 				if (strnlen(ext_addr, EXT_ADDRESS_LENGTH_CHAR+1) != EXT_ADDRESS_LENGTH_CHAR) {
-					fprintf(stderr, "%s: error: Wrong address length.%d \n", argv[0], strnlen(ext_addr, EXT_ADDRESS_LENGTH_CHAR+1));
+					fprintf(stderr, "%s: error: Wrong address length.%zu \n", argv[0], strnlen(ext_addr, EXT_ADDRESS_LENGTH_CHAR+1));
```
Comment (darconeous) [inline src/wpanctl/tool-cmd-commissioner.c]: For the sake of portability, I'd recommend keeping `%d` and just casting the `strlen` to `int`.

Comment (bukepo) [inline src/wpanctl/tool-cmd-commissioner.c]: >OpenThread uses and enforces the ISO9899:1999 (aka ISO C99, C99) C language standard as the minimum.

What does wpantund uses and enforces? **z** is in standard C99.

Comment (darconeous) [inline src/wpanctl/tool-cmd-commissioner.c]: The problem is that `strnlen` returns a `size_t`, which may be a `unsigned long long` or an `unsigned long` or even an `unsigned int`, depending on the platform. This means that there isn't a good universal way to handle the situation without casting.

Comment (bukepo) [inline src/wpanctl/tool-cmd-commissioner.c]: Is `z` modifier designed to solve this portability issue? If it's not allowed, shall we make it clear in `CONTRIBUTING.md` or other place that we should avoid new features introduced in C99?
I think other contributor may also confused about what to use and what not.

Comment (darconeous) [inline src/wpanctl/tool-cmd-commissioner.c]: Eh, C99 is fine. I guess my old pre-C99 instincts are kicking in. This is fine.

**Pull request merged**